src/os/bluestore/KernelDevice.cc

```diff
@@ -444,17 +455,20 @@ int KernelDevice::aio_zero(
   assert(len > 0);
   assert(off < size);
   assert(off + len <= size);
-
-  bufferlist bl;
-  while (len > 0) {
-    bufferlist t;
-    t.append(zeros, 0, MIN(zeros.length(), len));
-    len -= t.length();
-    bl.claim_append(t);
+  if (discard_zeroes_data) {
+    return block_device_discard(fd_direct, off, len);
```

Comment:
I'm not sure whether we need to write zero to get the goal. But zero actually is difference from discard, so if we only need discard here, I suggest we rename BlockDevice::aio_zero to BlockDevice::aio_discard, and add another aio_zero interface if necessary.

Like nvme spec provide zero command and discard command